src/System.Runtime.Extensions/src/System/IO/Path.Windows.cs

```diff
@@ -125,7 +125,25 @@ public static bool IsPathRooted(string path)
                 PathInternal.CheckInvalidPathChars(path);
 
                 int length = path.Length;
-                if ((length >= 1 && PathInternal.IsDirectorySeparator(path[0])) || 
+                if ((length >= 1 && PathInternal.IsDirectorySeparator(path[0])) ||
+                    (length >= 2 && path[1] == VolumeSeparatorChar))
+                    return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Tests if the given path starts with a root that is not equal to the 
+        /// forward slash.
+        /// </summary>
+        public static bool IsNonSeparatorRooted(string path)
+        {
+            if (path != null)
+            {
+                PathInternal.CheckInvalidPathChars(path);
+
+                int length = path.Length;
+                if ((length >= 1 && (path[0] == '\\')) ||
```

Comment:
>That will handle \\Server\Share, \\?\C:\, etc.

True, but it will also change the behavior of the single backslash case e.g.
```Path.Combine("/first", "\second")``` will return ```"/first\second"``` whereas currently it returns ```"\second"```.

So far I've only changed the forward-slash case. I'm not against changing the single backslash behavior but I expected it to be harder to push through.